### PR TITLE
fix: data source variables not working

### LINKED ISSUE #372: octopusdeploy_variables data source is not usable?
**Describe the bug**
The `octopusdeploy_variables` data source does not appear to be usable. The source for for [data_source_variables ](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/data_source_variables.go) seems to lack the patterns/conventions used in other data sources in this provider:
- non-Computed parameters used for querying ([example](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/schema_environment.go#L62-L64)) 
- Flattening Octopus API results back into the output data ([example](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/schema_environment.go#L35))

**Steps to reproduce**
1. Use a `octopusdeploy_variables` block in your terraform file as per https://registry.terraform.io/providers/OctopusDeployLabs/octopusdeploy/latest/docs/data-sources/variables
2. Note that you cannot set any parameters for querying for variables, as none are defined in the schema: https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/schema_variable.go#L66-L68
3. When Terraform executes the `octopusdeploy_variables` block, it attempts to read in values from fields that are not defined in the schema and/or are `Computed: true`. https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/master/octopusdeploy/data_source_variables.go#L20-L21

**Expected behavior**
Should be able to read Octopus variables from an Octopus Server installation and query by name, variable set id, etc.

**Logs and other supporting information**
- When no parameters supplied to `octopusdeploy_variables` block:
```
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-octopusdeploy_v0.7.69.exe plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 30 [running]:
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.dataSourceVariableReadByName({0x1044098, 0xc00007ca00}, 0x19a97ac1c68, {0xda5340, 0xc0000d9180})
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/data_source_variables.go:29 +0x577
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00026d5e0, {0x1044098, 0xc00007ca00}, 0xc0001a4300, {0xda5340, 0xc0000d9180})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:358 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc00026d5e0, {0x1044098, 0xc00007ca00}, 0xc000548080, {0xda5340, 0xc0000d9180})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:569 +0xf7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000696048, {0x1044098, 0xc00007ca00}, 0xc00020c1c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:1133 +0x3a8
```

- When `owner_id` is supplied to the block:
```
│ Error: Unsupported argument
│
│   on foo.tf line 16, in data "octopusdeploy_variables" "myvars":
│   16:   owner_id = "test-data"
│
│ An argument named "owner_id" is not expected here.
```

**Additional context**
When I use a locally modified version of the provider and expand the `data_source_variables` file by copying patterns/conventions used in the other data sources I am able to get the `octopusdeploy_variables` data source to function as I expect. I am wondering if the implementation for this particular data source was just missed or if I am not understanding/using it correctly.

### PR DESCRIPTION
**Pull request by domenicsim1**
closes #372

### CODE REVIEW

File: octopusdeploy/schema_variable.go [hunk 1]
```diff
@@ -55,19 +55,107 @@ func expandVariable(d *schema.ResourceData) *variables.Variable {
...
+			Type:        schema.TypeString,
+		},
+		"scope": {
+			Description: "As variable names can appear more than once under different scopes, a VariableScope must also be provided",
+			Elem:        &schema.Resource{Schema: getVariableScopeSchema()},
+			Required:    true,
 			Type:        schema.TypeList,
+			MaxItems:    1,
+		},
+		"space_id": getQuerySpaceID(),
```
Comment (benPearce1) [inline octopusdeploy/schema_variable.go:81]: does this mean the user cannot supply the space_id?

Comment (benPearce1) [reply octopusdeploy/schema_variable.go:81]: nvm, it seems that it was the same as before

**Review by benPearce1 — APPROVED** [review]
not tested, seems good

File: octopusdeploy/schema_variable.go (hunk 1 above)
Comment (domenicsim1) [reply octopusdeploy/schema_variable.go:81]: this means they can use it :)

**Pull request merged**